### PR TITLE
whirlpool-gui: init at 0.10.1

### PR DESCRIPTION
**Pull request by offlinehacker**
###### Motivation for this change

Add `whrilpool-gui` app.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/blockchains/whirlpool-gui/default.nix [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    '';
+  };
+
+  passthru.prefetchYarnCache = stdenv.lib.overrideDerivation yarnCache (d: {
+    outputHash = stdenv.lib.fakeSha256;
+  });
+
+  meta = {
+    description = "Desktop GUI for Whirlpool by Samourai-Wallet";
+    homepage = https://www.samouraiwallet.com/whirlpool;
```
Comment (mmahut) [inline pkgs/applications/blockchains/whirlpool-gui/default.nix:100]:
```suggestion
    homepage = "https://www.samouraiwallet.com/whirlpool";
```
Just cosmetic.

File: pkgs/applications/blockchains/whirlpool-gui/default.nix [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    extraEntries = ''
+      StartupWMClass=whrilpool-gui
+    '';
+  };
+
+  passthru.prefetchYarnCache = stdenv.lib.overrideDerivation yarnCache (d: {
+    outputHash = stdenv.lib.fakeSha256;
+  });
+
+  meta = {
```
Comment (mmahut) [inline pkgs/applications/blockchains/whirlpool-gui/default.nix:98]:
```suggestion
  meta = with stdenv.lib; {
```

**Comment by mmahut** [comment]
Tested, works fine, thank you.

**Comment by offlinehacker** [comment]
Hmm looks like I forgot to put url in quotes as is required according to rfc.

**Comment by mmahut** [comment]
> Hmm looks like I forgot to put homepage in quotes as is required according to rfc.

Yes, I did comment on it, but was not sure if it is already in RFC. :)

**Pull request merged**